qualtran/bloqs/qubitization_walk_operator.ipynb

```diff
@@ -28,6 +42,78 @@
     "import cirq"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "399690ef",
+   "metadata": {
+    "cq.autogen": "SelectOracle.bloq_doc.md"
+   },
+   "source": [
+    "## `SelectOracle`\n",
+    "Abstract base class that defines the interface for a SELECT Oracle.\n",
+    "\n",
+    "The action of a SELECT oracle on a selection register $|l\\rangle$ and target register\n",
+    "$|\\Psi\\rangle$ can be defined as:\n",
+    "\n",
+    "$$\n",
+    "    \\mathrm{SELECT} = \\sum_{l}|l \\rangle \\langle l| \\otimes U_l\n",
+    "$$\n",
+    "\n",
+    "In other words, the `SELECT` oracle applies $l$'th unitary $U_{l}$ on the target register\n",
+    "$|\\Psi\\rangle$ when the selection register stores integer $l$.\n",
+    "\n",
+    "$$\n",
+    "    \\mathrm{SELECT}|l\\rangle |\\Psi\\rangle = |l\\rangle U_{l}|\\Psi\\rangle\n",
+    "$$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b2bae756",
+   "metadata": {
+    "cq.autogen": "SelectOracle.bloq_doc.py"
+   },
+   "outputs": [],
+   "source": [
+    "from qualtran.bloqs.select_and_prepare import SelectOracle"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "774308d5",
+   "metadata": {
+    "cq.autogen": "PrepareOracle.bloq_doc.md"
+   },
+   "source": [
+    "## `PrepareOracle`\n",
+    "Abstract base class that defines the API for a PREPARE Oracle.\n",
+    "\n",
+    "Given a set of coefficients $\\{c_0, c_1, ..., c_{N - 1}\\}$, the PREPARE oracle is used to encode\n",
```

Comment:
nit: should probably be $\ell$ or $l$ here to match SELECT docs.

Reply:
$l$ refers to any given index not the final one though, right?

Reply:
Sorry I clicked the wrong line for this comment:  I meant the next line where you use $\sum_i$

Reply:
AH yes